nats.go

```diff
@@ -2026,33 +2018,32 @@ func (nc *Conn) Request(subj string, data []byte, timeout time.Duration) (*Msg,
 	}
 
 	// snapshot
-	var doSetup, useOldRequestStyle bool
 	nc.mu.Lock()
-	useOldRequestStyle = nc.Opts.UseOldRequestStyle
-	doSetup = (nc.respMux == nil)
+	useOldRequestStyle := nc.Opts.UseOldRequestStyle
 	nc.mu.Unlock()
 
 	// If user wants the old style.
 	if useOldRequestStyle {
 		return nc.oldRequest(subj, data, timeout)
 	}
 
-	// Make sure scoped subscription is setup at least once on first
-	// call to Request(). Will handle duplicates in createRespMux.
-	if doSetup {
-		if err := nc.createRespMux(); err != nil {
-			return nil, err
-		}
+	// Make sure scoped subscription is setup only once on first call to
+	// Request().
+	var err error
+	nc.respSetup.Do(func() { err = nc.createRespMux() })
```

Comment:
I think sync.Once() is great, but we could leave the use of it only when we detect that the respMux is not yet created. So it could be a mix of what we do in c-nats and use of sync.Once. 

That is, when first grabbing nc' mutex, we could setup the respMap, etc.. if not yet initialized, and prepare the respInbox and put it in the map, then snapshot if we need to create the subsription: `createSub := nc.respMux == nil`.  When releasing the nc's mutex, check `createSub` and invoke sync.Once, but this time just to create the subscription (not the map, which would already be done), then in the function, setup respMux under nc.mu protection please.

Again, in this case that means that we would grab nc.mu only once per request, except at initialization time where we may call sync.Once from multiple go-routines (if there are parallel Requests).

(Alternatively, or if I am not making sense, you could simply fix the map leak on timeout condition - which I had addressed in the c client and wanted to report back but did not have time, and then have I could submit a PR for the rest if it makes it easier to understand - and see if what I say actually works :-)).